includes/Core/Util/Tracking.php

```diff
@@ -24,56 +19,24 @@
  */
 final class Tracking {
 
-	/**
-	 * Tracking Optin Key
-	 *
-	 * @var string tracking optin key for options table.
-	 */
-	const TRACKING_OPTIN_KEY = 'googlesitekit_tracking_optin';
-
 	const TRACKING_ID = 'UA-130569087-3';
 
 	/**
-	 * Plugin context.
-	 *
-	 * @since 1.0.0
-	 * @var Context
-	 */
-	private $context;
-
-	/**
-	 * Authentication instance.
+	 * Tracking_Consent instance.
 	 *
-	 * @since 1.0.0
-	 * @var Authentication
+	 * @var Tracking_Consent
 	 */
-	protected $authentication;
-
-	/**
-	 * User_Options instance.
-	 *
-	 * @var User_Options
-	 */
-	protected $user_options;
+	protected $consent;
 
 	/**
 	 * Constructor.
 	 *
-	 * @since 1.0.0
-	 * @since 1.3.0 Added User_Options.
+	 * @since n.e.x.t
 	 *
-	 * @param Context        $context        Plugin context.
-	 * @param Authentication $authentication Optional. Authentication instance. Default is a new instance.
-	 * @param User_Options   $user_options   Optional. User_Options instance. Default is a new instance.
+	 * @param Tracking_Consent $consent Tracking_Consent instance.
 	 */
-	public function __construct(
-		Context $context,
-		Authentication $authentication = null,
-		User_Options $user_options = null
-	) {
-		$this->context        = $context;
-		$this->authentication = $authentication ?: new Authentication( $this->context );
-		$this->user_options   = $user_options ?: new User_Options( $this->context );
+	public function __construct( Tracking_Consent $consent ) {
+		$this->consent = $consent;
```

Comment:
I think I'd prefer to leave the constructor parameter-less for now. Unless there's an actual benefit to using DI here _already_, let's instantiate the class property unconditionally here, making it consistent with how `Authentication` instantiates its option/user option objects.